itests/manual_onboarding_test.go

```diff
@@ -0,0 +1,534 @@
+		sealedCid,
+		unsealedCid,
+		cacheDirPath,
+		sealedSectorPath,
+		sectorNumber,
+		actorId,
+		sealTickets,
+		seedRandomness,
+		[]abi.PieceInfo{},
+	)
+	req.NoError(err)
+
+	t.Logf("Running SealCommitPhase2 for sector %d...", sectorNumber)
+
+	sectorProof, err := ffi.SealCommitPhase2(scp1, sectorNumber, actorId)
+	req.NoError(err)
+
+	return sectorProof
+}
+
+func manualOnboardingGenerateWindowPost(
+	ctx context.Context,
+	t *testing.T,
+	client api.FullNode,
+	cacheDirPath string,
+	sealedSectorPath string,
+	minerAddr address.Address,
+	sectorNumber abi.SectorNumber,
+	sealedCid cid.Cid,
+) []byte {
+
+	req := require.New(t)
+
+	head, err := client.ChainHead(ctx)
+	req.NoError(err)
+
+	minerInfo, err := client.StateMinerInfo(ctx, minerAddr, head.Key())
+	req.NoError(err)
+
+	di, err := client.StateMinerProvingDeadline(ctx, minerAddr, types.EmptyTSK)
+	req.NoError(err)
+
+	minerAddrBytes := new(bytes.Buffer)
+	req.NoError(minerAddr.MarshalCBOR(minerAddrBytes))
+
+	rand, err := client.StateGetRandomnessFromBeacon(ctx, crypto.DomainSeparationTag_WindowedPoStChallengeSeed, di.Challenge, minerAddrBytes.Bytes(), head.Key())
+	req.NoError(err)
+	postRand := abi.PoStRandomness(rand)
+	postRand[31] &= 0x3f // make fr32 compatible
+
+	privateSectorInfo := ffi.PrivateSectorInfo{
+		SectorInfo: proof.SectorInfo{
+			SealProof:    kit.TestSpt,
+			SectorNumber: sectorNumber,
+			SealedCID:    sealedCid,
+		},
+		CacheDirPath:     cacheDirPath,
+		PoStProofType:    minerInfo.WindowPoStProofType,
+		SealedSectorPath: sealedSectorPath,
+	}
+
+	actorIdNum, err := address.IDFromAddress(minerAddr)
+	req.NoError(err)
+	actorId := abi.ActorID(actorIdNum)
+
+	windowProofs, faultySectors, err := ffi.GenerateWindowPoSt(actorId, ffi.NewSortedPrivateSectorInfo(privateSectorInfo), postRand)
+	req.NoError(err)
+	req.Len(faultySectors, 0)
+	req.Len(windowProofs, 1)
+	req.Equal(minerInfo.WindowPoStProofType, windowProofs[0].PoStProof)
+	proofBytes := windowProofs[0].ProofBytes
+
+	info := proof.WindowPoStVerifyInfo{
+		Randomness:        postRand,
+		Proofs:            []proof.PoStProof{{PoStProof: minerInfo.WindowPoStProofType, ProofBytes: proofBytes}},
+		ChallengedSectors: []proof.SectorInfo{{SealProof: kit.TestSpt, SectorNumber: sectorNumber, SealedCID: sealedCid}},
+		Prover:            actorId,
+	}
+
+	verified, err := ffi.VerifyWindowPoSt(info)
+	req.NoError(err)
+	req.True(verified, "window post verification failed")
+
+	return proofBytes
+}
+
+func manualOnboardingDisputeWindowPost(
+	ctx context.Context,
+	t *testing.T,
+	client kit.TestFullNode,
+	minerAddr address.Address,
+	sectorNumber abi.SectorNumber,
+) {
+
+	req := require.New(t)
+
+	head, err := client.ChainHead(ctx)
+	req.NoError(err)
+
+	sp, err := client.StateSectorPartition(ctx, minerAddr, sectorNumber, head.Key())
+	req.NoError(err)
+
+	di, err := client.StateMinerProvingDeadline(ctx, minerAddr, head.Key())
+	req.NoError(err)
+
+	disputeEpoch := di.Challenge + miner14.WPoStDisputeWindow + 5
```

Comment:
```
// WPoStDisputeWindow is the period after a challenge window ends during which
// PoSts submitted during that period may be disputed.
```

Now with 

```
disputeEpoch := di.Challenge + miner14.WPoStDisputeWindow + 5
```
This makes sense as we want to ensure that the WdPost message submitted by the Miner has achieved finality before we dispute it. But why do we use `di.Challenge` here instead of `di.Open`?